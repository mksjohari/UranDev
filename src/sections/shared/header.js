import React from "react";
import { Link } from "gatsby";

import styles from "../../modules/header.module.scss";
import Logo from "../../../static/logo.png";

const Header = () => {
    return (
        <header className={styles.header}>
            <div className={styles.logoAndDetails}>
                <div className={styles.logo}>
                    <Link className={styles.headerLink} to="/">
                        <img
                            className={styles.imageLogo}
                            src={Logo}
                            alt="Uran Logo ..."
                        />
                        <h1 className={styles.textLogo}>URAN</h1>
                    </Link>
                </div>
                <div className={styles.details}>
                    <div className={styles.signUp}>Sign Up</div>
                    <div className={styles.login}>Login</div>
                </div>
            </div>
            <div className={styles.navContainer}>
                <nav>
                    <ul className={styles.navList}>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/"
                            >
                                Home
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/explore"
                            >
                                Explore
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/projects"
                            >
                                Projects
                            </Link>
                        </li>
                        <li>
                            <Link
                                activeClassName={styles.activeNavItem}
                                className={styles.navItem}
                                to="/people"
                            >
                                People
                            </Link>
                        </li>
                    </ul>
                </nav>
            </div>
        </header>
    );
};

export default Header;
