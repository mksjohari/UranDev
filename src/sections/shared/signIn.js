import React, { useState } from "react";
import firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import { Link } from "gatsby";
import { Button } from "@material-ui/core";
import styles from "../../modules/signIn.module.scss";
import { getFirebase } from "../../firebase";
export const googleProvider = new firebase.auth.GoogleAuthProvider();

const signInWithGoogle = async () => {
    console.log("Sign in");
    getFirebase(firebase)
        .auth()
        .signInWithPopup(googleProvider)
        .then(function (result) {
            console.log(result);
        })
        .catch(function (error) {
            // Handle Errors here.
            const errorCode = error.code;
            const errorMessage = error.message;
            // The email of the user's account used.
            const email = error.email;
            // The firebase.auth.AuthCredential type that was used.
            const credential = error.credential;
            console.log(email, credential, errorCode, errorMessage);
            // ...
        });
};

const SignIn = () => {
    const [email, setEmail] = useState("");
    const [password, setPassword] = useState("");
    const [error] = useState(null);

    const signInWithEmailAndPasswordHandler = (event, email, password) => {
        event.preventDefault();
        console.log(event, email, password);
    };

    const onChangeHandler = (event) => {
        const { name, value } = event.currentTarget;

        if (name === "userEmail") {
            setEmail(value);
        } else if (name === "userPassword") {
            setPassword(value);
        }
    };
    console.log(getFirebase(firebase).auth().currentUser);
    return (
        <div className={styles.loginContainer}>
            <div className={styles.loginForm}>
                <h1>Sign In</h1>
                {error !== null && <div>{error}</div>}
                <form>
                    <label className={styles.label} htmlFor="userEmail">
                        Email:
                    </label>
                    <br />

                    <input
                        type="email"
                        name="userEmail"
                        value={email}
                        placeholder="E.g: uran@gmail.com"
                        id="userEmail"
                        onChange={(event) => onChangeHandler(event)}
                    />
                    <br />
                    <label className={styles.label} htmlFor="userPassword">
                        Password:
                    </label>
                    <br />

                    <input
                        type="password"
                        name="userPassword"
                        value={password}
                        placeholder="Your Password"
                        id="userPassword"
                        onChange={(event) => onChangeHandler(event)}
                    />
                    <br />
                    <Button
                        style={{ marginTop: 10 }}
                        color="primary"
                        variant="contained"
                        onClick={(event) => {
                            signInWithEmailAndPasswordHandler(
                                event,
                                email,
                                password
                            );
                        }}
                    >
                        Sign in
                    </Button>
                </form>
                <p className="text-center my-3">
                    Don't have an account? <Link to="/">Sign up here</Link>{" "}
                    <br /> <Link to="/">Forgot Password?</Link>
                </p>
                <hr />
                <Button
                    color="secondary"
                    variant="contained"
                    onClick={() => {
                        console.log("hello");
                        signInWithGoogle();
                    }}
                >
                    Sign in with Google
                </Button>
                {"   "}
                <Button
                    color="default"
                    variant="contained"
                    onClick={() => {
                        console.log(getFirebase(firebase).auth().currentUser);
                    }}
                >
                    Test Log
                </Button>
            </div>
        </div>
    );
};
export default SignIn;
