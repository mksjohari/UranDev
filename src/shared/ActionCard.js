import React from "react";
import { Draggable } from "react-beautiful-dnd";

import Options from "./sampleOptions";
import Dropdown from "./sandbox/Dropdown";
import styles from "../modules/tmp.module.scss";


class ActionCard extends React.Component {

  handleClick(event) {
    const task = document.getElementById(event.target.id.replace(/_editBtn/g,'') + '_edit');
    const type = event.target.className;

    if (type == "fas fa-edit") {
      task.style.display = "block";
      event.target.className = "fas fa-save";

      return;

    } else if (type == "fas fa-save") {
      task.style.display = "none";
      event.target.className = "fas fa-edit";

      return;
    }
  }


  render() {
    return (
      <Draggable 
        draggableId={this.props.action.id} 
        index={this.props.index}
      >
        {(provided) => (
          <div 
            {...provided.draggableProps}
            ref={provided.innerRef}
            className={styles.actionsCard}
          >
            <span className={styles.actionsContent}>
              <div className={styles.actions}>

              </div>

              <div className={styles.editAction} id={`${this.props.action.id}_edit`}>
                <div className={styles.tools} >
                  Used (tools)
                  <Dropdown 
                    options={Options} 
                    isMulti={true} 
                    isCreatable={true} 
                    keyColour="#AFDEE9"
                  />
                </div>
                <br />
                <div className={styles.tools} >
                  For (skills)
                  <Dropdown 
                    options={Options} 
                    isMulti={true} 
                    isCreatable={true}
                  />
                </div>
                <br />
                <div className={styles.actionDesc} >
                  Description (Optional)
                  <input className="inp-text"
                    placeHolder="type..."
                  ></input>
                </div>
              </div>
            </span>
            <div className={styles.btnCol}>
              <span {...provided.dragHandleProps} className={styles.dragHandle}>
                <i class="fas fa-grip-lines"></i>
              </span>
              <span className={styles.editBtn}>
                <i 
                  class="fas fa-edit" 
                  id={`${this.props.action.id}_editBtn`} 
                  onClick={this.handleClick}
                ></i>
              </span>
            </div>
          </div>
        )}
      </Draggable>
    );
  }
}

export default ActionCard;
