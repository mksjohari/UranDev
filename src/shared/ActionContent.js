import React, { Component } from "react";
import styles from "../modules/tmp.module.scss";

class ActionContent extends Component {


  render() {
    return (
      <>
        <div>{this.props.tools[0] ? 'Used tools' : ''}</div>
        <div className={this.props.tools[0] ? styles.tools : ''}>
          {this.props.tools[0] ?
            (this.props.tools.map((tool) => (
              <span className={styles.toolItem}>{tool}</span>
              )
            )) : <></>
          } 
        </div>
        <div>{this.props.skills[0] ? 'Used skills' : ''}</div>
        <div className={this.props.skills[0] ? styles.skills : ''}>
          {this.props.skills[0] ?
            (this.props.skills.map((skill) => (
              <span className={styles.skillItem}>{skill}</span>
              )
            )) : <></>
          } 
        </div>
        <div>{this.props.description.length ? 'Description:' : ''}</div>
        <div className={styles.description}>{this.props.description}</div>
      </>
    );
  }
}

export default ActionContent;
