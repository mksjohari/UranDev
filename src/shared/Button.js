import React from "react";

function Button(props) {
  return (
    <>
      <button
        type="button"
        className={props.colour}
        id={props.id}
        onClick={props.onClick}
      >
        {props.iconL ? props.iconL : ""}
        <span>{props.text}</span>
        {props.iconR ? props.iconR : ""}
      </button>
    </>
  );
}

export default Button;
