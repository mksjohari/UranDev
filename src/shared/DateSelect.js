import React from 'react';
//import ThemedStyleSheet from 'react-with-styles/lib/ThemedStyleSheet';
//import aphroditeInterface from 'react-with-styles-interface-aphrodite';
//import DefaultTheme from 'react-dates/lib/theme/DefaultTheme';

import 'react-dates/initialize';
import 'react-dates/lib/css/_datepicker.css';
import '../modules/react_dates_overrides.css'
import { DateRangePicker, SingleDatePicker, DayPickerRangeController } from 'react-dates';

// ThemedStyleSheet.registerInterface(aphroditeInterface);
// ThemedStyleSheet.registerTheme({
//   reactDates: {
//     ...DefaultTheme.reactDates,
//     color: {
//       ...DefaultTheme.reactDates.color,
//       highlighted: {
//         backgroundColor: 'lightblue',
//         backgroundColor_active: 'lightblue',
//         backgroundColor_hover: 'lightblue',
//         color: 'lightblue',
//         color_active: 'lightblue',
//         color_hover: 'lightblue',
//       },
//     },
//   },
// });

class DateSelect extends React.Component {
  state = {};
  
  render() {
    return(
      <>
        <DateRangePicker
          startDate={this.state.startDate} // momentPropTypes.momentObj or null,
          startDateId={this.props.taskId + "_startDate"} // PropTypes.string.isRequired,
          endDate={this.state.endDate} // momentPropTypes.momentObj or null,
          endDateId={this.props.taskId + "_endDate"} // PropTypes.string.isRequired,
          onDatesChange={({ startDate, endDate }) => this.setState({ startDate, endDate })} // PropTypes.func.isRequired,
          focusedInput={this.state.focusedInput} // PropTypes.oneOf([START_DATE, END_DATE]) or null,
          onFocusChange={focusedInput => this.setState({ focusedInput })} // PropTypes.func.isRequired,
          disableScroll={true}
          showClearDates
          reopenPickerOnClearDates
          customInputIcon={<i className={`fas fa-calendar`} ></i>}
          small
          minimumNights={0}
          isOutsideRange={() => false}
        />
      </>
    )
  }
}

export default DateSelect;
