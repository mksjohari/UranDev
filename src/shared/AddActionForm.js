import React from 'react';

import Dropdown from './sandbox/Dropdown';
import Options from './sampleOptions';
import Button from './sandbox/Button';
import Droparea from './sandbox/Droparea';

import styles from '../modules/addActionForm.module.scss';

class AddActionForm extends React.Component {
  state = {};

  handleChange(event) {

    var newDesc = event.target.value.replace(/\n/g, "");
    event.target.value = newDesc;
    
    const newState = {
      ...this.state,
      action: {
        ...this.state.action,
        description: newDesc,
      }
    };

    this.setState(newState);
  }

  resetHeight(event) {
    event.target.style.height = 1 + "px";
    event.target.style.height = (5 + event.target.scrollHeight) + "px";
    // console.log(event.target.scrollHeight);
  }

  handleChange = this.handleChange.bind(this);

  render() {
    return (
      <>
        <div className={styles.container}>
          <h3>Add Action</h3>
          <div className={styles.toolsEdit} >
            <div className={styles.title} >What tool(s) did you use?</div>
            <Dropdown 
              id={this.props.actionid + '_tools'}
              options={Options.tools} 
              isMulti={true} 
              isCreatable={true} 
              colour="white"
              // defaultValue={this.state.action.tools.map(
              //   (tool) => (
              //     { value: tool, label: tool }
              //   )) }
              keyColour="#AFDEE9"
              onChange={this.handleSelectTool}
            />
          </div>
          <br />
          <div className={styles.skillsEdit} >
            <div className={styles.title} >What Skill(s) allowed you to achieve this?</div>
            <Dropdown 
              id={this.props.actionid + '_skills'}
              options={Options.skills} 
              isMulti={true} 
              isCreatable={true}
              colour="white"
              // defaultValue={this.state.action.skills.map(
              //   (skill) => (
              //     { value: skill, label: skill }
              //   )) }
              onChange={this.handleSelectSkill}
            />
          </div>
          <br />
          <div className={styles.actionDesc} >
            <div className={styles.title} >Describe your action in detail. (Optional)</div>
            <textarea 
              id={this.props.actionid + '_desc'}
              className={styles.descEdit}
              placeholder='Type...'
              onChange={this.handleChange}
              onKeyUp={this.resetHeight}
            ></textarea>
          </div>
          <br/>
          <div className={styles.actionFiles} >
            <div className={styles.title} >Add files to showcase your work in progress. (Optional, max 10 files)</div>
            <Droparea className={styles.actionFiles} />
          </div>

          <div className={styles.btnRow} >
            <Button
              colour='pink'
              id={this.props.id + '_add'}
              text="Confirm"
              iconR={<i className="fas fa-check" ></i>}
              onClick={this.props.addAction}
            />
            <Button
              colour='reddo'
              id={this.props.id + '_close'}
              text="Cancel"
              iconR={<i className="fas fa-times" ></i>}
              onClick={this.props.close}
            />
          </div>

        </div>
      </>
    )
  }
}

export default AddActionForm;
