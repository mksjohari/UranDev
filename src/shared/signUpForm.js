import React, { useState } from 'react';
import { connect } from 'react-redux';

import { Link, useHistory } from 'react-router-dom';
import Button from './sandbox/Button';
import PasswordStrengthMeter from '../shared/sandbox/PasswordStrengthMeter';
import '../modules/loginform.scss';
import Loader from './Loader';

import { getFirebase } from './firebase/config';
import { signInWithGoogle } from './signInForm';
import { checkUserExists, createAccount } from './firebase/firebase';
import { updateInfo, updateInfoFromSignUp } from '../actions/userAction';

// const sendVerification = () => {
//     const user = getFirebase().auth().currentUser;
//     user.sendEmailVerification()
//         .then(() => {
//             console.log("Verification Email Sent");
//         })
//         .catch((err) => {
//             console.log("Error Occured");
//             console.log(err);
//         });
// };

const SignUp = React.memo((props) => {
	const [email, setEmail] = useState('');
	const [password, setPassword] = useState('');
	const [confirmPass, setConfirmPass] = useState('');
	const [firstName, setFirstName] = useState('');
	const [lastName, setLastName] = useState('');
	const [errormsg, setErrormsg] = useState('');
	const [svgIndex, setIndex] = useState(-1);
	const [svgOpacity, setOpacity] = useState(0);
	let history = useHistory();
	const handleSubmit = (e) => {
		e.preventDefault();
		setIndex(1);
		setOpacity(90);
		if (password.length < 8) {
			setErrormsg('Password must be at least 8 characters');
		} else if (password === confirmPass) {
			getFirebase()
				.auth()
				.createUserWithEmailAndPassword(email, password)
				.then(async (user) => {
					const exists = await checkUserExists(user.user.uid);
					const uuid = user.user.uid;
					if (exists.data[0] === false) {
						createAccount({
							uuid: uuid,
							firstName,
							lastName,
							email,
						}).then(() => {
							props.updateInfoFromSignUp({
								uuid,
								firstName,
								lastName,
								email,
							});
							props.onClose();
							history.push('/signup', {
								fromSignUp: true,
								firstName,
								lastName,
							});
						});
						console.log('Created account', uuid);
					} else {
						console.log('Account already exists');
					}
				})
				.catch(function (error) {
					// Handle Errors here.
					var errorCode = error.code;
					var errorMessage = error.message;
					if (errorCode === 'auth/email-already-in-use') {
						console.log(errorMessage);
						setErrormsg(
							'This email address is already in use. Please log in instead.'
						);
					} else {
						console.log(error);
					}
					// ...
				});
		} else {
			setErrormsg("The passwords don't match. Try again.");
		}
	};
	return (
		<form className="login-form" onSubmit={handleSubmit}>
			<Loader
				loadMessage="Creating your account"
				style={{
					position: 'absolute',
					backgroundColor: 'white',
					opacity: `${svgOpacity}%`,
					zIndex: `${svgIndex}`,
					padding: '210px 50px',
				}}
			/>

			<i className="fas fa-times" onClick={props.onClose} />
			<h2 style={{ margin: '0 0 12px 0' }}>Sign Up</h2>
			<input
				className="inp-text"
				placeholder="First Name"
				type="text"
				onChange={(e) => setFirstName(e.target.value)}
				required
			/>
			<input
				className="inp-text"
				placeholder="Last Name"
				type="text"
				onChange={(e) => setLastName(e.target.value)}
				required
			/>
			<input
				className="inp-text"
				placeholder="&#xf0e0;   Email"
				type="text"
				onChange={(e) => setEmail(e.target.value)}
				required
			/>
			<span className="divider">
				<hr style={{ width: '100%' }} />
			</span>
			<input
				className={!errormsg ? 'inp-text' : 'inp-text inp_error'}
				placeholder="&#xf070;   Password"
				type="password"
				onChange={(e) => setPassword(e.target.value)}
				required
			/>
			<input
				className="inp-text"
				placeholder="&#xf070;   Confirm Password"
				type="password"
				onChange={(e) => setConfirmPass(e.target.value)}
				required
			/>
			<PasswordStrengthMeter password={password} />
			{errormsg && <p className="error">{errormsg}</p>}
			<input
				className="pink login button"
				id="login"
				type="submit"
				value="Create account"
			/>
			<span className="divider">
				<hr style={{ width: '100px' }} />
				<span className="small-text">or</span>
				<hr style={{ width: '100px' }} />
			</span>
			<Button
				type="button"
				iconL={<i className="fab fa-google-plus-g" />}
				id="google"
				text="Sign up with Google"
				onClick={() => {
					signInWithGoogle(
						props.onClose,
						history,
						props.updateInfo,
						setIndex,
						setOpacity
					);
				}}
			/>
			<span className="small-text">
				Already have an account?{' '}
				<Link
					className="link"
					to="/"
					onClick={() => props.toSignIn()}
					style={{ textAlign: 'right' }}
				>
					Sign in.
				</Link>
			</span>
		</form>
	);
});

export default connect(null, { updateInfo, updateInfoFromSignUp })(SignUp);
