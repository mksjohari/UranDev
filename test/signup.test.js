import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
const renderer = require("react-test-renderer");
configure({ adapter: new Adapter() });
import { SignUp } from "../src/shared/signUpForm";

describe("<SignUp /> component runs", () => {
    it("SignUp component renders without crashing", () => {
        shallow(<SignUp />);
    });
    it("SignUp component matches previous snapshots", () => {
        const tree = renderer
            .create(
                <BrowserRouter>
                    <SignUp />
                </BrowserRouter>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe("<SignUp /> component renders correctly", () => {
    let container = null;
    beforeAll(() => {
        // setup a DOM element as a render target
        container = document.createElement("div");
        document.body.appendChild(container);
        ReactDOM.render(
            <BrowserRouter>
                <SignUp />
            </BrowserRouter>,
            container
        );
    });

    afterAll(() => {
        // cleanup on exiting
        ReactDOM.unmountComponentAtNode(container);
        container.remove();
        container = null;
    });
    it("Renders <form>", () => {
        expect(document.querySelector("form").className).toBe("login-form");
        expect(document.querySelector("h2").innerHTML).toBe("Sign Up");
    });

    it("Renders 5 <input> fields", () => {
        expect(document.getElementsByClassName("inp-text")).toHaveLength(5);
    });
    it("Renders correct input types and placeholders", () => {
        expect(document.getElementsByClassName("inp-text")[0].placeholder).toBe(
            "First Name"
        );
        expect(document.getElementsByClassName("inp-text")[1].placeholder).toBe(
            "Last Name"
        );
        expect(document.getElementsByClassName("inp-text")[2].type).toBe(
            "text"
        );
        expect(document.getElementsByClassName("inp-text")[(3, 4)].type).toBe(
            "password"
        );
    });
    it("Renders 2 <button>s", () => {
        expect(document.getElementById("login").type).toBe("submit");
        expect(document.getElementById("google").type).toBe("button");
    });
});

describe("<SignUp /> component handles interactions", () => {
    const user = {
        firstName: "John",
        lastName: "Doe",
        email: "johndoe@gmail.com",
        password: "test1234",
        confirmPass: "test12345",
    };
    const handleSubmit = jest.fn();
    const wrapper = mount(
        <BrowserRouter>
            <SignUp onSubmit={handleSubmit} />
        </BrowserRouter>
    );
    it("Form handles input state changes when onChange functions are invoked", () => {
        const inputs = wrapper.find(".inp-text");
        inputs.at(0).simulate("change", {
            target: {
                value: user.firstName,
            },
        });
        inputs.at(1).simulate("change", {
            target: {
                value: user.lastName,
            },
        });
        inputs.at(2).simulate("change", {
            target: {
                value: user.email,
            },
        });
        inputs.at(3).simulate("change", {
            target: {
                value: user.password,
            },
        });
        inputs.at(4).simulate("change", {
            target: {
                value: user.confirmPass,
            },
        });
        expect(inputs.at(0).getDOMNode().value).toBe(user.firstName);
        expect(inputs.at(1).getDOMNode().value).toBe(user.lastName);
        expect(inputs.at(2).getDOMNode().value).toBe(user.email);
        expect(inputs.at(3).getDOMNode().value).toBe(user.password);
        expect(inputs.at(4).getDOMNode().value).toBe(user.confirmPass);
    });
    it("Form handles onSumbit prop", () => {
        // wrapper.instance().handleSubmit = handleSubmit;
        // const form = wrapper.find("form");
        // // console.log(form.props());
        // wrapper.find("form").simulate("submit", { preventDefault: () => {} });
        // expect(handleSubmit).toHaveBeenCalled();
    });
});

// const handleChange = jest.spyOn(React, "useState");
// handleChange.mockImplementation((errormsg) => [errormsg, setErrormsg]);
// expect(setErrormsg).toBe("The passwords don't match. Try again.");
// console.log(submit.debug());
