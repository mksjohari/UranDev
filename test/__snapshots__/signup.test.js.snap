// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> component runs without errors SignUp component matches previous snapshots 1`] = `
<form
  className="login-form"
  onSubmit={[Function]}
>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "opacity": "0%",
        "padding": "210px 50px",
        "position": "absolute",
        "zIndex": "-1",
      }
    }
  >
    <svg
      height="67.2px"
      id="loader"
      version="1.1"
      viewBox="0 0 60 67.2"
      width="60px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        className="st0"
        d="M41,50.9V21.3c0-2.7-2.2-4.9-4.9-4.9H6.4c-2.7,0-4.9,2.2-4.9,4.9v39.5c0,2.7,2.2,4.9,4.9,4.9h19.7L41,50.9z"
      />
      <polyline
        className="st0"
        points="41,50.9 26.2,50.9 26.2,65.7 "
      />
      <line
        className="st0"
        id="line1"
        x1="11.4"
        x2="31.1"
        y1="38.6"
        y2="38.6"
      />
      <line
        className="st0"
        id="line2"
        x1="11.4"
        x2="31.1"
        y1="28.7"
        y2="28.7"
      />
      <polyline
        className="st1"
        id="line3"
        points="11.4,48.5 13.8,48.5 16.3,48.5 "
      />
      <path
        className="st2"
        d="M52,2.6c1.5-1.5,3.9-1.5,5.5,0c1.5,1.5,1.5,3.9,0,5.5L39,26.5l-7.5,2l2-7.5L52,2.6z"
        id="pencil"
      />
      <line
        className="st2"
        id="pencil"
        x1="49.6"
        x2="55.1"
        y1="5"
        y2="10.4"
      />
    </svg>
    <br />
    <div
      className="loading"
    >
      Creating your account
       
      <div
        id="dot1"
      />
      <div
        id="dot2"
      />
      <div
        id="dot3"
      />
    </div>
  </div>
  <i
    className="fas fa-times"
  />
  <h2
    style={
      Object {
        "margin": "0 0 12px 0",
      }
    }
  >
    Sign Up
  </h2>
  <input
    className="inp-text"
    onChange={[Function]}
    placeholder="First Name"
    required={true}
    type="text"
    value=""
  />
  <input
    className="inp-text"
    onChange={[Function]}
    placeholder="Last Name"
    required={true}
    type="text"
    value=""
  />
  <input
    className="inp-text"
    onChange={[Function]}
    placeholder="   Email"
    required={true}
    type="text"
    value=""
  />
  <span
    className="divider"
  >
    <hr
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </span>
  <input
    className="inp-text"
    onChange={[Function]}
    placeholder="   Password"
    required={true}
    type="password"
    value=""
  />
  <input
    className="inp-text"
    onChange={[Function]}
    placeholder="   Confirm Password"
    required={true}
    type="password"
    value=""
  />
  <div
    className="password-strength-meter"
  >
    <progress
      className="password-strength-meter-progress strength-Weak"
      max="4"
      value={0}
    />
    <br />
    <label
      className="password-strength-meter-label"
    >
      
    </label>
  </div>
  
  <input
    className="pink login button"
    id="login"
    type="submit"
    value="Create account"
  />
  <span
    className="divider"
  >
    <hr
      style={
        Object {
          "width": "100px",
        }
      }
    />
    <span
      className="small-text"
    >
      or
    </span>
    <hr
      style={
        Object {
          "width": "100px",
        }
      }
    />
  </span>
  <button
    className="undefined undefined"
    id="google"
    onClick={[Function]}
    style={
      Object {
        "display": "inline",
      }
    }
    type="button"
  >
    <i
      className="fab fa-google-plus-g"
    />
    <span>
      Sign up with Google
    </span>
    
    
  </button>
  <span
    className="small-text"
  >
    Already have an account?
     
    <a
      className="link"
      href="/"
      onClick={[Function]}
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      Sign in.
    </a>
  </span>
</form>
`;
